sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// Registers an <see cref="IPersistableModel{T}"/> proxy to be used when reading or writing a model.
+    /// </summary>
+    /// <param name="proxy"> The <see cref="IPersistableModel{T}"/> proxy that will be used to read or write the model. </param>
+    public void AddProxy<T>(IPersistableModel<T> proxy)
+    {
+        if (_proxies is null)
+            _proxies = new Dictionary<Type, object>();
+
+        _proxies.Add(typeof(T), proxy);
+    }
+
+    /// <summary>
+    /// Gets the model that is currently being proxied.
```

Comment:
Consider the following clarification:

```suggestion
    /// Gets the instance of a model that is currently being read or written via a proxy at that stage of reading or writing the model graph.
```

Reply:
I think this makes it more confusing.  Currently being proxied sounds clearer to me.

Reply:
The words in the comment could imply that you return the model type, not the model instance.  It would be helpful to users to disambiguate this.